src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+interrupt at a lower privilege mode since the higher-privilege mode
+causes the interrupt signal to appear more urgent than any lower-privilege
+mode interrupt.
+
+NOTE: This register is defined as WARL as some implementations may want to support a limited number
+of values in this register including hardwiring some bits to fixed values.
+
+NOTE: Within a single privilege mode, it can be useful to separate interrupt
+handler tasks from application tasks to increase robustness, reduce
+space usage, and aid in system debugging.  Interrupt handler tasks
+have non-zero interrupt levels, while application tasks have an
+interrupt level of zero.
+
+=== New {mtvec} CSR Mode for CLIC
+
+The CLIC interrupt-handling mode is encoded as a new state in the
+existing {mtvec} WARL register, where {mtvec}.`mode` (the two
+least-significant bits) is `11`.
+
+.CLIC mode xtvec register layout
+include::images/wavedrom/xtvec.edn[]
+
+NOTE: CLINT mode in this specification is defined as `mtvec.mode=00` or `mtvec.mode=01`.
+
+[source]
+----
+ mode  PC on interrupt
+ ====  ============================================
+ 00       OBASE              # CLINT direct mode
+ 01       OBASE+4*exccode    # CLINT vectored mode
+ 11       NBASE              # CLIC mode
+ 10       Reserved
+
+where:
+  OBASE = xtvec[XLEN-1:2]<<2   # CLINT mode vector base is at least 4-byte aligned
+  NBASE = xtvec[XLEN-1:2]<<2   #  CLIC mode vector base is at least 4-byte aligned
+----
+
+In CLIC mode, synchronous exception traps always jump to NBASE.
+
+If the hart is currently running at some privilege mode (`pp`) at some
+interrupt level (`pil`) and an enabled interrupt becomes pending at
+any interrupt level in a higher privilege mode or if an interrupt assigned to a
+higher interrupt level in the current privilege mode becomes pending
+and interrupts are globally enabled in this privilege mode, then
+execution is immediately transferred to a handler running with the new
+interrupt's privilege mode (`**__x__**`) and interrupt level (`il`).
+
+==== New Interrupt Status ({mpintstatus}) CSR
+
+A new M-mode CSR, `mpintstatus`, holds consolidated state of preempted context.
+
+NOTE: Previous versions of CLIC modified the {cause} CSR to contain preempted context.
+That modification has been moved to the smcliclat extension.
+{pintstatus} has been added to allow implementations to maintain {cause} compatibility between
+CLIC and CLINT modes.
+
+[source]
+----
+mpintstatus 
+Bits   Field         Description
+ XLEN-1 Interrupt     Set to 1 for interrupts and 0 for exceptions.  Mirror of mcause.Interrupt
+ 30     (reserved)    Contains minhv (AKA xinhv) bit when smclicshv extension present
+ 29:28  mpp[1:0]      Previous privilege mode, mirror of mstatus.mpp
+ 27     mpie          Previous interrupt enable, mirror of mstatus.mpie
+ 26:24  (reserved)
+ 23:16  mpil[7:0]     Previous interrupt level
+ 15:12  (reserved)
+ 11:0   exccode[11:0] Exception/interrupt code, mirror of mcause[11:0]

```

Comment:
I am not sure we need exccode here. We should discuss this separately I believe. If anything, we would want the previous exccode. However, since we are dropping nxti (thats my assumption), and will advertise for push/pop, i think the problem we were facing might go away.

Reply:
https://github.com/riscv/riscv-fast-interrupt/issues/467